doc/project/detailed_contribution_guide/_index.md

```diff
@@ -0,0 +1,340 @@
+
+## Coding style guides, formatting tools
+
+All source code contributions must adhere to project style guides.
+We use separate style guides for different languages:
+* [SystemVerilog](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md)
+* [C/C++]({{< relref "c_cpp_coding_style.md" >}})
+* [Python]({{< relref "python_coding_style.md" >}})
+* [Markdown]({{< relref "markdown_usage_style.md" >}})
+* [Hjson]({{< relref "hjson_usage_style.md" >}})
+* [RISC-V Assembly]({{< relref "asm_coding_style.md" >}})
+
+If unsure about the style, be consistent with the existing code and do your best to match its style.
+
+For C/C++ code, we recommend to run "git-clang-format" on your changes.
+
+## What should commit messages look like?
+
+There are many [useful guides](https://chris.beams.io/posts/git-commit/) available on how to write commit messages and why.
+
+The following list contains a set of simple guidelines that we consider most useful.
+* A commit message consists of a subject line followed by an optional body.
+* Keep the subject line short. Ideally under 50 characters.
+* Capitalize the subject line.
+* Use the imperative mood in the subject line.
+* Use the present tense ("Add feature" not "Added feature").
+* When changes are restricted to a specific area, you should add a tag to the start of the subject line in square brackets, e.g. "\[uart/rtl\] Rework Tx FSM". See also [subject tags](#subject-tags).
+* Do not end the subject line with a period.
+* Separate subject from body with a blank line.
+* Wrap the body at 72 characters.
+* Use the body to explain what and why (rather than how).
+
+To indicate acceptance of our Contributor License Agreement (CLA), commits should also include a Signed-off-by line (which can be generated by `git commit -s`).
+See [CONTRIBUTING.md](https://github.com/lowRISC/opentitan/blob/master/CONTRIBUTING.md) for more information.
+
+## Subject tags
+
+If code changes, issues or pull requests (PRs) are restricted to a specific area, we recommend adding a tag in square brackets, *e.g.*, "\[uart/rtl\]", to the subject line of your commit message, issue or PR.
+Such tags help to identify areas affected by changes.
+Tags are particularly helpful when dealing with pull requests (PRs) as they can guide team members for code review and simplify later retrieval of specific PRs through the GitHub web interface.
+
+You can add multiple tags separated by a comma, e.g., "\[top, dv\]" to indicate that multiple separated areas are affected.
+In addition, you can add further levels of hierarchy to make the indication more precise such as "\[uart/rtl\]" instead of just "\[uart\]", "\[sw/device\]" instead of "\[sw\]".
+
+## What about TODO comments?
+
+We track non-trivial pieces of work via GitHub issues rather than TODO comments in the code.
+We do not use TODO(name) comments as used in some code bases - if it is important enough to assign someone, it is worth tracking via an issue.
```

Comment:
We have some instances of `FIXME` comments. I think it would be useful to add some guidance on how and when to use them.

Reply:
That's a good point. Options are:
* TODO/FIXME are broadly equivalent, people should just use TODO
* TODO/FIXME are broadly equivalent, but we don't mind if people use one or the other based on what seems most appropriate to them
* We define a split between TODO/FIXME. I don't know if there's a commonly used definition but [PEP0350](https://www.python.org/dev/peps/pep-0350/) has TODO as "Informal tasks/features that are pending completion." and FIXME as "Areas of problematic or ugly code needing refactoring or cleanup."

My initial thought is leaning more towards one of the first bullets. If there are lots of views on this, we should probably break it out to a separate issue to resolve.

What do you think?

Reply:
Option (2) has the advantage that it doesn't need changes to the existing codebase to become "conformant". I'd just suggest viewing them as synonyms.

Reply:
FWIW, in just the `*.[ch]` files, there are 106 TODOs and 19 FIXMEs.  Among them, most of the FIXMEs I would say should be TODOs.  IOW, people have been treating them as roughly equivalent.

I'd be ok with (2), but more in favor of (1): we should just use TODOs and work on eliminating the FIXMEs as part of the normal course of development (ie: don't go over the codebase and convert them as some house-cleaning effort, just deal with them in their own time, but don't add new FIXMEs).

Reply:
I've added "Code authors are encouraged to use TODO rather than FIXME." which I think reflects this discussion, and is hopefully non-contentious enough that we don't need to go back and forth on it in a separate issue thread.